src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -33,6 +60,7 @@ export const TodoFilter: React.FC = () => {
             data-cy="clearSearchButton"
             type="button"
             className="delete"
+            onClick={handleRemoveQuery}
```

Comment:
use conditional rendering to hide the button when it's needed:
![image](https://github.com/mate-academy/react_redux-list-of-todos/assets/86923089/6f7d51e4-6f25-48e6-a7c4-7544fd8d3f35)